### PR TITLE
Nested forms

### PR DESCRIPTION
**Pull request by bgrundmann**
So this just deals with nested forms as a way to compose complex forms by splitting functionality into several sub forms, and thereby possibly being able to reuse such sub forms across several different forms.

This also gives iommi something akin to https://docs.djangoproject.com/en/3.1/ref/forms/fields/#multivaluefield .

Unlike my previous announcements / previews it does *not* provide formsets yet.  Basically I haven't yet figured out how to do a combinator that creates a dynamic list of parts at bind time yet.  (Well I kinda did but the versions I have either make reinvoke angry or cause the "disturbance in the force error message", so not really).

But as per the example I think this is useful as is.

### CODE REVIEW

File: examples/examples/form_examples.py [hunk 1]
```diff
@@ -298,6 +302,45 @@ class Meta:
...
+            def post_validation(form, **_):
+                print("post validation", form.is_valid())
+                if form.is_valid():
+                    if form.fields.first_day.value > form.fields.last_day.value:
+                        form.add_error("First day must be <= last day")
+
+    class MyForm(Form):
+        event = Field()
+        # attr='' => instance.first_day, instance.last_day instead of instance.when.first_day, instance.when.last_day
+        when = DateRangeField(attr='')
```
Comment (boxed) [inline examples/examples/form_examples.py:326]: I didn't understand this comment or why `attr=''` is here..

Comment (boxed) [reply examples/examples/form_examples.py:326]: Ok, now I got it.. but since I didn't get it the first time, maybe we can come up with something better...

**Comment by boxed** [comment]
Does this mean that forms can be nested like a->b->c->d?

**Comment by bgrundmann** [comment]
Yes.  Well in iommi with this PR.  Not in web (as you know).  What happens in this PR is that the forms will turn into one outer form with all inner forms becoming divs.

The outer forms.is_valid and apply recurse as you would expect.  A inner form's is_valid only needs its children and fields to be valid, not the parent.  Which again is as you would expect I claim... ;-)

File: examples/examples/form_examples.py (hunk 1 above)
Comment (bgrundmann) [reply examples/examples/form_examples.py:326]: Yeah I wasn't super happy with that either.  Maybe a default of attr='' on Forms would be more correct...

**Pull request merged**